File: Ex02/Animals_JS/js/Bird.js
var Bird = function(name, carnivore, wingSpan){
	//calling super constructor
	Animal.call(this, name, carnivore);
	
	//double; oeffentliche EIgenschaft
	this.mWingSpan = wingSpan;
	
	
	//da abstrakte Klasse
	if(this.constructor === Bird){
		throw new Error ("Bird: Cannot instatiate abstract class");
		// console.log("Bird: Cannot instatiate abstract class");
	}
	
	
};
Bird.prototype = Object.create(Animal.prototype);
Bird.prototype.constructor = Bird;

//Elternklasse-Methoden

Bird.prototype.setWingSpan = function(wingSpan){
	
		this.mWingSpan = wingSpan;
	
}

Bird.prototype.getWingSpan = function(){
	
		return this.mWingSpan;
}

Bird.prototype.equals = function(birdOb){
	//check super class equality
	var animalEqality = Object.getPrototypeOf(Bird.prototype).equals(birdOb);
	console.log("anim equality im bird" + animalEquality);
	if(animalEquality == true)
	{
		if( (this.mWingSpan == birdOb.getWingSpan()) )
		{
			return true;
		}
		return false;
	}
	
	else
	{
		return false;
	}
	
 }
	
	
function createBird() {
	try{

		console.log("creating bird...");
		var m = new Bird ("Test" , "true");

		return m;
	}
	catch(e){
	
		console.log("Exception:" + e);
		return e;
	}
};

File: Ex02/Animals_ObjC/Bird.m
//
//  Bird.m
//  
//
//  Created by Elisabeth Hoppe on 25.05.15.
//
//

#import "Animal.h"
#import "Bird.h"

@implementation Bird:Animal

- (id)initWithParams: (NSMutableString*)name_ carnivore:(BOOL)carnivore_ wingSpan:(double)wingSpan_
{
    self=[super initWithParams:name_  carnivore:carnivore_];
    if(self){
        [self setWingSpan:wingSpan_];
    }
    return self;
}

- (void)setWingSpan: (double)wingSpan_
{
    wingSpan = wingSpan_;
}

- (double)getWingSpan
{
    return wingSpan;
}

- (void)toString
{
    //TODO sort
    NSLog(@"ID: %d Name: %@ Carnivore: %d Wingspan: %f", [self getID], [self getName], [self isCarnivore], [self getWingSpan] );
}

-(BOOL) equals:(Bird *)bird
{
    //check the super class equality
    BOOL animalEqualiy = [super equals:bird];
    
    if(animalEqualiy == YES)
    {
        //check for equality in own class
        if(wingSpan == [bird getWingSpan])
        {
            return YES;
        }
        else
        {
            return NO;
        }
    }
    else
    {
        return NO;
    }
    
    return NO;
    
}
@end
